src/realm/util/platform_info.hpp

```diff
@@ -36,6 +36,39 @@ inline std::string get_platform_info()
             info.machine); // Throws
 }
 
+inline std::string get_library_platform()
+{
+#if REALM_ANDROID
+    return "Android";
+#elif REALM_WINDOWS
+    return "Windows";
+#elif REALM_UWP
+    return "UWP";
+#elif REALM_IOS
+    return "iOS";
+#elif REALM_PLATFORM_APPLE
+    return "MacOS";
```

Comment:
Doesn't this obscure tvOS? And I guess watchOS, once we start supporting Sync there?

Reply:
@fealebenpae  Are you suggesting we introduce a `REALM_MACOS`?

Reply:
We already have `REALM_TVOS`, `REALM_MACCATALYST`, and `REALM_WATCHOS` we should be testing before we test for `REALM_PLATFORM_APPLE` and conclude that if it's none of the former then all that's left is `macOS`.

Reply:
Specifically I think this should read:
```
#elif REALM_MACCATALYST // test Catalyst first because it's a subset of iOS
  return "Mac Catalyst";
#elif REALM_IOS
  return "iOS";
#elif REALM_TVOS
  return "tvOS";
#elif REALM_WATCHOS
  return "watchOS";
#elif REALM_APPLE_PLATFORM
  return "macOS";
```